plugin/src/main/java/org/gradle/testretry/internal/config/TestTaskConfigurer.java

```diff
@@ -61,6 +64,30 @@ public static void configureTestTask(Test test, ObjectFactory objectFactory, Pro
         test.doLast(new ConditionalTaskAction(isDeactivatedByTestDistributionPlugin, new FinalizeTaskAction()));
     }
 
+    private static void ensureThatNoRetryExtensionIsPresent(Test testTask) {
+        Object existingRetryExtension = testTask.getExtensions().findByName(TestRetryTaskExtension.NAME);
+
+        if (existingRetryExtension != null) {
+            String retryExtensionClassName = existingRetryExtension.getClass().getName();
+
+            if (retryExtensionClassName.startsWith(GRADLE_ENTERPRISE_BASE_PACKAGE)) {
+                throw new IllegalStateException(""
+                    + "The Gradle Enterprise Gradle plugin is conflicting with the Test Retry Gradle plugin "
+                    + "and has already added a retry extension to the test task " + testTask.getName() + ". "
+                    + "Please either remove the Test Retry Gradle plugin from this project "
+                    + "or disable the registration of the retry extension in the Gradle Enterprise Gradle plugin "
+                    + "by specifying the system property 'gradle.enterprise.testretry.disabled'."
+                );
+            } else {
```

Comment:
❔ do we prevent double application of the retry plugin itself already elsewhere?